### PR TITLE
CORE-8711 Listing Groups in the Collaborators window

### PR DESCRIPTION
**Pull request by ash3rz**
I've migrated/consolidated most of the old the Groups code into the Collaborators package and refactored the view to look like we discussed in the latest mock up.  

I can either hide the new Collaborator List view or wait to merge this in until the Groups endpoints are in terrain.

NOTE: Will require ansible changes before merging.

image

### CODE REVIEW

File: de-lib/src/main/java/org/iplantc/de/collaborators/client/presenter/ManageCollaboratorsPresenter.java [hunk 1]
```diff
@@ -110,6 +115,26 @@ public void onFailure(Throwable caught) {
 
     }
 
+    void updateListView() {
+        String searchTerm = "*";
```
Comment (psarando) [inline de-lib/src/main/java/org/iplantc/de/collaborators/client/presenter/ManageCollaboratorsPresenter.java]: We really should be converting our "search" endpoints like this into "list" endpoints with an optional `search` query param (see cyverse-de/apps#47 for example).

File: de-lib/src/main/java/org/iplantc/de/collaborators/client/views/ManageCollaboratorsViewImpl.java [hunk 2]
```diff
@@ -119,10 +120,10 @@ public void loadData(List<Collaborator> models) {
     @Override
     public void mask(String maskText) {
         if (maskText == null || maskText.isEmpty()) {
```
Comment (psarando) [inline de-lib/src/main/java/org/iplantc/de/collaborators/client/views/ManageCollaboratorsViewImpl.java]: Here's an opportunity to use `Strings.isNullOrEmpty`.

File: de-lib/src/main/java/org/iplantc/de/collaborators/client/views/ManageCollaboratorsViewImpl.java [hunk 3]
```diff
@@ -119,10 +120,10 @@ public void loadData(List<Collaborator> models) {
...
-            collaboratorListPnl.mask(I18N.DISPLAY.loadingMask());
-        } else {
-            collaboratorListPnl.mask(maskText);
+            maskText = appearance.loadingMask();
```
Comment (psarando) [inline de-lib/src/main/java/org/iplantc/de/collaborators/client/views/ManageCollaboratorsViewImpl.java]: Tidy!✨

File: de-lib/src/main/resources/org/iplantc/de/resources/client/messages/IplantContextualHelpStrings.properties [hunk 4]
```diff
@@ -1,4 +1,4 @@
...
+collaboratorsHelp = <p>The Collaborators window allows you to quickly find, add, and remove CyVerse users within your set of Collaborators.</p><p>Adding someone as your collaborator enhances the sharing feature, making it even easier to share data with one, a few, or all of your collaborators.</p> <p>To add a collaborator, begin typing their name or CyVerse username in the search field and a results drop-down will appear. Select your desired name to add them to your set of collaborators. You can also easily access this set when sharing.</p><p>Additionally, if you have the need to frequently share data with several of the same collaborators, you can create a Collaborator List.  Switch over to the Collaborator Lists tab and click the Add button to start creating your own custom lists of Collaborators.</p>
```
Comment (psarando) [inline de-lib/src/main/resources/org/iplantc/de/resources/client/messages/IplantContextualHelpStrings.properties]: Is there any way to use line-breaks in one of these for readability?

File: de-lib/src/main/java/org/iplantc/de/collaborators/client/views/ManageCollaboratorsViewImpl.java (hunk 2 above)
Comment (ash3rz) [inline de-lib/src/main/java/org/iplantc/de/collaborators/client/views/ManageCollaboratorsViewImpl.java]: Whoops, good catch!

File: de-lib/src/main/resources/org/iplantc/de/resources/client/messages/IplantContextualHelpStrings.properties [hunk 5]
```diff
@@ -1,4 +1,7 @@
...
+collaboratorsHelp = <p>The Collaborators window allows you to quickly find, add, and remove CyVerse users within your set of Collaborators.</p>\
+  <p>Adding someone as your collaborator enhances the sharing feature, making it even easier to share data with one, a few, or all of your collaborators.</p>\
+  <p>To add a collaborator, begin typing their name or CyVerse username in the search field and a results drop-down will appear. Select your desired name to add them to your set of collaborators. You can also easily access this set when sharing.</p>\
+  <p>Additionally, if you have the need to frequently share data with several of the same collaborators, you can create a Collaborator List.  Switch over to the Collaborator Lists tab and click the Add button to start creating your own custom lists of Collaborators.</p>
```
Comment (psarando) [inline de-lib/src/main/resources/org/iplantc/de/resources/client/messages/IplantContextualHelpStrings.properties]: 👍

**Comment by ash3rz** [comment]
Thanks for the review!  I'll make the ansible changes first, then merge this in.

**Pull request merged**